pkg/sql/backfill.go

```diff
@@ -93,6 +93,14 @@ var indexBackfillMergeBatchSize = settings.RegisterIntSetting(
 	settings.NonNegativeInt, /* validateFn */
 )
 
+var indexBackfillMergeBatchBytes = settings.RegisterIntSetting(
+	settings.TenantWritable,
+	"bulkio.index_backfill.merge_batch_bytes",
+	"the max number of bytes we merge between temporary and adding indexes in a single batch",
+	16<<10,
```

Comment:
Where did this number come from? It feels small. Ideally what we'd do is both have this be larger and pull this much out of the memory monitor before issuing it. 

At the very least, comment how this number was chosen.

Reply:
If we want to choose based on some prior art, we use 16 MB over in changefeeds: https://github.com/cockroachdb/cockroach/blob/4fa089a2b6f11e1d46a8ea5d881dc4641d818b70/pkg/ccl/changefeedccl/kvfeed/scanner.go#L121. Although we run those scans in parallel which we don't do yet here.

I opened an issue for the memory monitor here: https://github.com/cockroachdb/cockroach/issues/77123

Reply:
Done, I'll use 16MiB here. It looks like it should be larger than 1000 KVs from looking at indexes in the demo. Also added the memory monitor and pull the `merge_batch_bytes` out of it before sending the request.